docs/basics/101-140-filesystem.rst

```diff
@@ -0,0 +1,764 @@
+
+
+To summarize, renaming files is easy and worry-free. Don't be intimidated
+by a file marked as deleted -- a :command:`datalad save` will rectify this.
+Be mindful of other modifications in your dataset, though, and either supply
+appropriate paths to ``datalad save``, or use Git tools to exclusively save
+the name change and nothing else.
+
+Let's revert this now, to have a clean history.
+
+.. runrecord:: _examples/DL-101-140-110
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git reset --hard HEAD~1
+   $ datalad status
+
+
+Moving files from or into subdirectories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's move an annexed file from within ``books/`` into the root
+of the superdataset:
+
+.. runrecord:: _examples/DL-101-140-120
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ mv TLCL.pdf ../TLCL.pdf
+   $ datalad status
+
+In general, this looks exactly like renaming or moving a file
+in the same directory. There is a subtle difference though:
+Currently, the symlink of the annexed file is broken. There
+are two ways to demonstrate this. One is trying to open the
+file -- this will currently fail. The second way is to look
+at the symlink:
+
+.. runrecord:: _examples/DL-101-140-121
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ cd ../
+   $ ls -l TLCL.pdf
+
+The first part of the symlink should point into the ``.git/``
+directory, but currently, it doesn't -- the symlink still looks
+like ``TLCL.pdf`` would be within ``books/``. Instead of pointing
+into ``.git``, it currently points to ``../.git``, which is non-existent,
+and even outside of the superdataset. This is why the file
+cannot be opened: When any program tries to follow the symlink,
+it will not resolve, and an error such as "no file or directory"
+will be returned.. But don't panic! A ``datalad save`` will
+rectify this as well:
+
+.. runrecord:: _examples/DL-101-140-122
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad save -m "moved book into root"
+   $ ls -l TLCL.pdf
+
+After a ``datalad save``, the symlink is fixed again.
+Therefore, in general, whenever moving or renaming a file,
+especially between directories, a ``datalad save`` is
+the best option to turn to.
+
+.. container:: toggle
+
+   .. container:: header
+
+      **Find out more:** Why a move between directories is actually a content change
+
+   Let's see how this shows up in the dataset history:
+
+   .. runrecord:: _examples/DL-101-140-123
+      :language: console
+      :workdir: dl-101/DataLad-101/books
+
+      $ git log -1 -p
+
+   As you can see, this action does not show up as a move, but instead
+   a deletion and addition of a new file. Why? Because the content
+   that is tracked is the actual symlink, and due to the change in
+   relative location, the symlink needed to change. Hence, what looks
+   and feels like a move on the file system for you is actually a
+   move plus a content change for Git.
```

Comment:
A pointer to the `git annex fix` command could be added to this box -- which is the one that is triggered by `datalad save` -> `git commit`, and actually does the job of fixing things up inside.